src/operators/arrange/arrangement.rs

```diff
@@ -589,14 +589,12 @@ where
                     // and sending smaller bites than we might have otherwise done.
 
                     // Assert that the frontier never regresses.
-                    assert!(input.frontier().frontier().iter().all(|t1| input_frontier.iter().any(|t2: &G::Timestamp| t2.less_equal(t1))));
-
-                    // Test to see if strict progress has occurred (any of the old frontier less equal
-                    // to the new frontier).
-                    let progress = input_frontier.iter().any(|t2| !input.frontier().less_equal(t2));
-
-                    if progress {
+                    assert!(PartialOrder::less_equal(&prev_frontier.borrow(), &input.frontier().frontier()));
 
+                    // Test to see if strict progress has occurred, which happens whenever the new
+                    // frontier isn't equal to the previous. It is only in this case that we have any
+                    // data processing to do.
+                    if prev_frontier.borrow() != input.frontier().frontier() {
```

Comment:
I think this is right. I was a bit confused about the definition of antichain equality, but it looks like the implementation equates two things that have the same elements but in different orders.